aten/src/ATen/Utils.h

```diff
@@ -127,4 +128,30 @@ inline int64_t prod_intlist(ArrayRef<int64_t> list) {
   return std::accumulate(list.begin(), list.end(), 1ll, std::multiplies<int64_t>());
 }
 
+/*
+* Utility function used in tensor implementations, which
+* supplies the default generator to tensors, if an input generator
+* is not supplied. The input Generator* is also dynamic casted to
+* the backend generator type (CPU/CUDAGenerator etc.)
+*/
+template <typename T>
+static inline T * check_generator_with_default(Generator * expr, Generator * defaultValue) {
+  if (!expr)
+    expr = defaultValue;
+  if(auto result = dynamic_cast<T*>(expr))
```

Comment:
We can't assume RTTI is available on all platforms we support. You should add a tag to the Generator superclass and use that to test for the types in question.